### PR TITLE
fix(EMS-880-896): Company Details - Multiple sic codes fix & goods and services max count fix

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issues:
* Sic codes were duplicated each time company details page was submitted
* Characters left was missing from goods and services field

# Changes made:
* Added helper to populate old sic codes in application
* API deletes old sic codes when they exist in application already
* Added maximum to goods and services page variables
* Added and updated tests

### CODE REVIEW

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js [hunk 1]
```diff
@@ -59,4 +68,16 @@ describe("Insurance - Your business - Company details page - As an Exporter I wa
       cy.url().should('eq', natureOfBusinessUrl);
     });
   });
+
+  describe('when resubmitting company number on company details page', () => {
+    before(() => {
+      partials.backLink().click();
+      submitButton().click();
+      partials.backLink().click();
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js:76]: Could this be simpler? Seems it's in a bit of a loop here - back, submit, back?

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js [hunk 2]
```diff
@@ -59,4 +68,16 @@ describe("Insurance - Your business - Company details page - As an Exporter I wa
...
+    });
+
+    it('it should remove old sic codes from company summary list', () => {
+      cy.checkText(partials.yourBusinessSummaryList[COMPANY_SIC].value(), '64999');
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js:80]: Can we pull the value in from mock-application?

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js:76]: I can navigate to url if that's easier, will be doing the same thing though

File: src/api/custom-schema.ts [hunk 3]
```diff
@@ -132,6 +137,12 @@ export const extendGraphqlSchema = (schema: GraphQLSchema) =>
 
             const mappedSicCodes = mapSicCodes(company, sicCodes);
 
+            if (mappedSicCodes && oldSicCodes) {
+              await context.db.ExporterCompanySicCode.deleteMany({
+                where: [...oldSicCodes],
+              });
+            }
+
```
Comment (ttbarnes) [inline src/api/custom-schema.ts:145]: Can we add some comments to this resolver now, lots going on 

Also do we need to do a `.length` check here?

File: src/ui/server/helpers/get-sic-code-ids-from-application/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe('when there are multiple sic codes in exporterCompany', () => {
+    it('should return a populated array array', () => {
+      const applicationWithMultipleSicCodes = { ...mockApplication };
+      applicationWithMultipleSicCodes.exporterCompany.sicCodes.push({
+        id: 'clcyyxldc0634m8novkr94sss',
+        sicCode: '649955',
```
Comment (ttbarnes) [inline src/ui/server/helpers/get-sic-code-ids-from-application/index.test.ts:19]: Can we pull in from test mocks?

File: src/api/.keystone/config.js [hunk 5]
```diff
@@ -538,6 +543,11 @@ var extendGraphqlSchema = (schema) => (0, import_schema.mergeSchemas)({
             data: address
           });
           const mappedSicCodes = mapSicCodes(company, sicCodes);
+          if (mappedSicCodes && oldSicCodes) {
+            await context.db.ExporterCompanySicCode.deleteMany({
+              where: [...oldSicCodes]
```
Comment (ttbarnes) [inline src/api/.keystone/config.js:548]: Looks like we don't need to spread here?

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js:76]: Could it not just be one line of `partials.backLink().click();` ?

Comment (Zainzzkk) [reply e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/company-details-continue-no-errors.spec.js:76]: it is on the nature of the business page so I need to get back to company details, and then it redirects again so need to go back again

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**